pool/app/pool_user/event.ml

```diff
@@ -7,6 +8,7 @@ type event =
   | Unblocked of t
   | PasswordUpdated of
       Id.t * Password.Plain.t * Password.Plain.t * Password.Confirmation.t [@opaque]
+  | PasswordReset of string * Password.Plain.t * Password.Confirmation.t [@opaque]
```

Comment:
Please use the Token type for the event:

```suggestion
  | PasswordReset of Pool_token.t * Password.Plain.t * Password.Confirmation.t [@opaque]
```

Update Pool_token to expose it's type instead of `string`:

```ocaml
(* pool_token.ml *)
type t = (string[@opaque]) [@@deriving eq, show]

let value m = m
```


```ocaml
(* pool_token.mli *)
type t

val equal : t -> t -> bool
val pp : Format.formatter -> t -> unit
val show : t -> string
val value : t -> string
```

- [ ] update all pool_token functions to take `Pool_token.t` instead of `string`